nnvm/python/nnvm/testing/check_computation.py

```diff
@@ -0,0 +1,546 @@
+# pylint: disable=cell-var-from-loop,no-else-return
+"""Helper utilities to check functions and their gradients."""
+from __future__ import absolute_import as _abs
+
+import logging
+import numpy as np
+
+import tvm
+from tvm.contrib import graph_runtime
+
+import nnvm
+from nnvm.compiler import graph_util
+from nnvm.compiler.graph_attr import TCODE_TO_DTYPE
+from .config import ctx_list
+
+def graph_to_function(graph, target, ctx, shape=None, dtype=None):
+    """Convert a graph to a function taking a keyword args and returning a list of results
+    (both args and results are numpy arrays).
+
+    Example::
+
+        fun = graph_to_function(graph, llvm, cpu(0))
+        [res1, res2] = fun(x=np.zeros((1,2)), y=np.zeros((1,)))
+
+    Parameters
+    ----------
+    graph : nnvm.graph.Graph
+        A graph we want to convert to a function.
+
+    target : str or :any:`tvm.target.Target`
+        The build target
+
+    ctx : TVMContext
+        The context to deploy the module.
+
+    shape : Dict[str, Tuple[int]], optional
+        A dict mapping input variable names to shapes.
+        By default shapes will be inferred from variables' attributes.
+        Note that this parameter takes precedence over variables' attributes.
+
+    dtype : Dict[str, str] or str, optional
+        A dict mapping input variable names to dtypes, or just a single dtype.
+        By default dtypes will be inferred from variables' attributes.
+        Note that this parameter takes precedence over variables' attributes.
+
+    Returns
+    -------
+    function : Callable[..., List[numpy.ndarray]]
+    """
+
+    if shape is None or dtype is None:
+        all_shapes = graph.json_attr('shape')
+        all_dtypes = graph.json_attr('dtype')
+
+        if all_shapes is None or all_dtypes is None:
+            graph = graph.apply('InferShape').apply('InferType')
+            all_shapes = graph.json_attr('shape')
+            all_dtypes = graph.json_attr('dtype')
+
+        all_dtypes = [None if t == -1 else TCODE_TO_DTYPE[t] for t in all_dtypes]
```

Comment:
TCODE_TO_DTYPE[-1] is defined as None in graph_attr.py.  The if-else expression is not necessary.